src/Geometry.cpp

```diff
@@ -55,25 +66,92 @@ Geometry::Geometry( const std::string filename, GLenum r, Shader* s1, Shader* s2
 
             normals.push_back( glm::vec3( nom.x, nom.y, nom.z ) );
         }
+
+        // With aiProcess_Triangulate, we are guaranteed that face.mNumIndices = 3
+        for ( uint j = 0; j < mesh->mNumFaces; ++j ) {
+            aiFace face = mesh->mFaces[j];
+
+            for ( uint k = 0; k < face.mNumIndices; ++k ) {
+                faces.push_back( face.mIndices[k] );
+            }
+        }
     }
 
     InitializeVAO();
     Load();
 }
 
-Geometry::Geometry( std::vector<glm::vec3> v, std::vector<glm::vec3> c, GLenum r )
+Geometry::Geometry( std::vector<glm::vec3> v, std::vector<glm::vec3> c, std::vector<glm::vec3> n, GLenum r, Shader* geo, Shader* stencil )
     : vertexBuffer( 0 )
     , textureBuffer( 0 )
     , colourBuffer( 0 )
     , normalBuffer( 0 )
+    , faceBuffer( 0 )
     , vertexArray( 0 )
     , renderMode( r )
-    , program( nullptr )
-    , programOutline( nullptr )
+    , program( geo )
+    , programOutline( stencil )
     , vertices( std::move( v ) )
     , colours( std::move( c ) )
-    , normals( {} )
-, ModelMatrix( glm::mat4( 1.0f ) ) {
+    , normals( std::move( n ) )
+    , ModelMatrix( glm::mat4( 1.0f ) ) {
+
+    for ( int i = 0; i < vertices.size() / 3; ++i ) {
```

Comment:
for ( uint i = 0; i < vertices.size() ; i+=3 ) {
       faces.push_back( i );
       faces.push_back( i + 1 );
      faces.push_back( i + 2 );
   }